llvm/docs/TableGen/ProgRef.rst

```diff
@@ -1275,6 +1275,23 @@ be nested.
 This loop defines records named ``R0``, ``R1``, ``R2``, and ``R3``, along
 with ``F0``, ``F1``, ``F2``, and ``F3``.
 
+``dump`` --- print messages to stderr
+-------------------------------------
+
+A ``dump`` statement prints the input string to standard error
```

Comment:
It may be useful to clarify when the printout actually happens. 
E.g. for `dump` inside a class, that would probably be during individual record instantiation.

This suggests that the printout order will not match that of the order of `dump()` in the source file. I wonder if we should print out some sort of location context -- e.g. the record or class name. Just a suggestion. I haven't tried it yet and don't know how usable it is in practice.

Reply:
Done, docs updated.